sklearn/manifold/t_sne.py

```diff
@@ -253,10 +258,12 @@ def _kl_divergence_bh(params, P, degrees_of_freedom, n_samples, n_components,
     error = _barnes_hut_tsne.gradient(val_P, X_embedded, neighbors, indptr,
                                       grad, angle, n_components, verbose,
                                       dof=degrees_of_freedom,
-                                      compute_error=compute_error)
+                                      compute_error=compute_error,
+                                      num_threads=num_threads)
     c = 2.0 * (degrees_of_freedom + 1.0) / degrees_of_freedom
     grad = grad.ravel()
-    grad *= c
+    if c != 1:
```

Comment:
does it really help? (curious)

Reply:
I guess it does as it avoids a potentially large multiplication and `c = 1` is the most common case. But I am not sure as I did the implementation quite some time ago. I would say at least it does not hurt. :)

Reply:
The most common case is `degrees_of_freedom = 1`, which leads to `c = 4`, isn't it ?

Reply:
Indeed.. Then, I don't know why I did that, probably some debugging.
I will revert.